extensions/Lily/SoundExpanded.js

```diff
@@ -0,0 +1,355 @@
+      soundPlayer.outputNode.loop = true;
+    }
+
+    stopLooping(args, util) {
+      const index = this._getSoundIndex(args.SOUND, util);
+      if (index < 0) return false;
+      const sprite = util.target.sprite;
+
+      const soundId = sprite.sounds[index].soundId;
+      const soundPlayer = sprite.soundBank.soundPlayers[soundId];
+
+      if (!soundPlayer.outputNode) return;
+      soundPlayer.outputNode.loop = false;
+    }
+
+    isLooping(args, util) {
+      const index = this._getSoundIndex(args.SOUND, util);
+      if (index < 0) return false;
+      const sprite = util.target.sprite;
+
+      const soundId = sprite.sounds[index].soundId;
+      const soundPlayer = sprite.soundBank.soundPlayers[soundId];
+
+      if (!soundPlayer.outputNode) return false;
+      return soundPlayer.outputNode.loop;
+    }
+
+    isSoundPlaying(args, util) {
+      const index = this._getSoundIndex(args.SOUND, util);
+      if (index < 0) return false;
+      const sprite = util.target.sprite;
+
+      const soundId = sprite.sounds[index].soundId;
+      const soundPlayers = sprite.soundBank.soundPlayers;
+      return soundPlayers[soundId].isPlaying;
+    }
+
+    attributeOfSound(args, util) {
+      const index = this._getSoundIndex(args.SOUND, util);
+      if (index < 0) return 0;
+      const sprite = util.target.sprite;
+
+      const soundId = sprite.sounds[index].soundId;
+      const soundPlayer = sprite.soundBank.soundPlayers[soundId];
+      const soundBuffer = soundPlayer.buffer;
+
+      switch (args.ATTRIBUTE) {
+        case ('length'):
+          return Math.round(soundBuffer.duration * 100) / 100;
+        case ('channels'):
+          return soundBuffer.numberOfChannels;
+        case ('sample rate'):
+          return soundBuffer.sampleRate;
+      }
+    }
+
+    getSoundEffect(args, util) {
+      const target = getSpriteTargetByName(util, args.TARGET);
+      const effects = target.soundEffects;
+      if (!effects) return 0;
+      return effects[args.EFFECT];
+    }
+
+    /* Utility Functions */
+
+    _getSoundIndex(soundName, util) {
+      const len = util.target.sprite.sounds.length;
+      if (len === 0) {
+        return -1;
+      }
+      const index = this._getSoundIndexByName(soundName, util);
+      if (index !== -1) {
+        return index;
+      }
+      const oneIndexedIndex = parseInt(soundName, 10);
+      if (!isNaN(oneIndexedIndex)) {
+        return this._wrapClamp(oneIndexedIndex - 1, 0, len - 1);
+      }
+      return -1;
+    }
+
+    _getSoundIndexByName(soundName, util) {
+      const sounds = util.target.sprite.sounds;
+      for (let i = 0; i < sounds.length; i++) {
+        if (sounds[i].name === soundName) {
+          return i;
+        }
+      }
+      return -1;
+    }
+
+    _wrapClamp(n, min, max) {
+      const range = (max - min) + 1;
+      return n - (Math.floor((n - min) / range) * range);
+    }
+
+    _getTargets() {
+      const spriteNames = [];

```

Comment:
let's avoid the same mistake as looks+. have the top value always be "myself" with value `_myself_` and just list every sprite under it, and also add the stage (`_stage_`)

Reply:
I made a script that does that for skins, I'll add that now